cmd/pillard/tests/setup_test.go

```diff
@@ -0,0 +1,55 @@
+// Package tests implements users tests for the API layer.
+package tests
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/ardanlabs/kit/cfg"
+	"github.com/ardanlabs/kit/db"
+	"github.com/ardanlabs/kit/web/app"
+
+	"github.com/coralproject/xenia/cmd/pillard/routes"
+	"github.com/coralproject/xenia/tstdata"
+)
+
+var a *app.App
+
+func init() {
+	// The call to API will force the init() function to initialize
+	// cfg, log and mongodb.
+	a = routes.API(true).(*app.App)
+}
+
+//==============================================================================
+
+// TestMain helps to clean up the test data.
+func TestMain(m *testing.M) {
+	os.Exit(runTest(m))
+}
+
+// runTest initializes the environment for the tests and allows for
+// the proper return code if the test fails or succeeds.
+func runTest(m *testing.M) int {
+	// In order to get a Mongo session we need the name of the database we
+	// are using. The web framework middleware is using this by convention.
+	dbName, err := cfg.String("MONGO_DB")
+	if err != nil {
+		fmt.Println("MongoDB is not configured")
+		return 1
+	}
+
+	db, err := db.NewMGO("context", dbName)
+	if err != nil {
+		fmt.Println("Unable to get Mongo session")
+		return 1
+	}
+
+	defer db.CloseMGO("context")
+
```

Comment:
This is a testing harness, individual test for pkgs to be stored in individual pkgs along with fixtures and triggered here.  See https://github.com/coralproject/xenia/blob/master/cmd/xeniad/tests/setup_test.go#L57 for usage of this pattern fro xeniad.